api/types/src/address.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_types::account_address::AccountAddress;
+use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt, str::FromStr};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Address(AccountAddress);
+
+impl Address {
+    pub fn new(address: AccountAddress) -> Self {
+        Self(address)
+    }
+
+    pub fn into_inner(&self) -> AccountAddress {
+        self.0
+    }
+}
+
+impl fmt::Display for Address {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0.to_hex_literal())
+    }
+}
+
+impl FromStr for Address {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self, anyhow::Error> {
+        Ok(Self(AccountAddress::from_hex_literal(s).map_err(|_| {
+            anyhow::format_err!("invalid account address: {}", s)
```

Comment:
Why do you need to create another error when `from_hex_literal` already returns an error, I'd recommend simply using that one

Reply:
the one from the from_hex_literal does not have invalid address info.